clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/SerializeApplicationAtomicOperation/SerializeApplicationAtomicOperation.groovy

```diff
@@ -0,0 +1,671 @@
+      }
+      if (disk.deviceName) {
+        diskMap.device_name = disk.deviceName
+      }
+      if (disk.initializeParams?.diskName) {
+        diskMap.name = disk.initializeParams.diskName
+      }
+      if (disk.initializeParams?.sourceImage) {
+        diskMap.source_image = disk.initializeParams.sourceImage
+      }
+      if (disk.source) {
+        diskMap.source = disk.source
+      }
+      if (disk.interface) {
+        diskMap.interface = disk.interface
+      }
+      if (disk.mode) {
+        diskMap.mode = disk.mode
+      }
+      if (disk.initializeParams?.diskType) {
+        diskMap.disk_type = disk.initializeParams?.diskType
+      }
+      if (disk.initializeParams?.diskSizeGb) {
+        diskMap.disk_size_gb = disk.initializeParams?.diskSizeGb
+      }
+      if (disk.type) {
+        diskMap.type = disk.type
+      }
+      instanceTemplateMap.disk << diskMap
+      numDisks++
+    }
+    return numDisks > 0
+  }
+
+  private Void addAutoscalerToResourceMap(String targetName, String targetZone, AutoscalingPolicy autoscalingPolicy, Map resourceMap) {
+    def autoscalerMap = [:]
+
+    autoscalerMap.name = targetName
+    autoscalerMap.target = "\${google_compute_instance_group_manager.${targetName}.self_link}"
+    autoscalerMap.zone = targetZone
+
+    // Autoscaling policy
+    autoscalerMap.autoscaling_policy = [:]
+    if (autoscalingPolicy.maxNumReplicas) {
+      autoscalerMap.autoscaling_policy.max_replicas = autoscalingPolicy.maxNumReplicas
+    } else {
+      throw new GoogleResourceImproperFormat("Required maximum number of replicas not found for autoscaler: ${autoscalerMap.name}")
+    }
+    if (autoscalingPolicy.minNumReplicas) {
+      autoscalerMap.autoscaling_policy.min_replicas = autoscalingPolicy.minNumReplicas
+    } else {
+      throw new GoogleResourceImproperFormat("Required minimum number of replicas not found for autoscaler: ${autoscalerMap.name}")
+    }
+    if (autoscalingPolicy.coolDownPeriodSec) {
+      autoscalerMap.autoscaling_policy.cooldown_period = autoscalingPolicy.coolDownPeriodSec
+    }
+    //TODO(nwwebb) terraform only lets you select one policy even though gce lets you select multiple
+    if (autoscalingPolicy.cpuUtilization?.utilizationTarget) {
+      autoscalerMap.autoscaling_policy.cpu_utilization = [:]
+      autoscalerMap.autoscaling_policy.cpu_utilization.target = autoscalingPolicy.cpuUtilization.utilizationTarget
+    }
+    if (autoscalingPolicy.customMetricUtilizations) {
+      autoscalerMap.autoscaling_policy.metric = []
+      autoscalingPolicy.customMetricUtilizations.each {Map metric ->
+        def metricMap = [:]
+        if (metric.metric) {
+          metricMap.name = metric.metric
+        } else {
+          return
+        }
+        if (metric.utilizationTarget) {
+          metricMap.target = metric.utilizationTarget
+        } else {
+          return
+        }
+        //TODO(nwwebb) ask about this, gce doesn't match terraform types
```

Comment:
Another good time to file an issue: https://github.com/hashicorp/terraform/issues